renku/core/workflow/converters/renku.py

```diff
@@ -0,0 +1,138 @@
+from renku.core.util.yaml import write_yaml
+from renku.domain_model.workflow.composite_plan import CompositePlan
+from renku.domain_model.workflow.converters import IWorkflowConverter
+from renku.domain_model.workflow.plan import Plan
+
+
+class RenkuWorkflowFileExporter(IWorkflowConverter):
+    """Export a ``CompositePlan`` or a ``Plan`` to Renku workflow file."""
+
+    @hookimpl
+    def workflow_format(self):
+        """Workflow format name."""
+        return self, ["renku"]
+
+    @hookimpl
+    def workflow_convert(
+        self,
+        workflow: Union[CompositePlan, Plan],
+        basedir: Path,
+        output: Optional[Path],
+        output_format: Optional[str],
+        resolve_paths: Optional[bool],
+        nest_workflows: Optional[bool],
+    ) -> str:
+        """Converts the specified workflow to renku workflow file."""
+
+        def get_output_path():
+            if output:
+                path = output / workflow.name if output.is_dir() else output
+            else:
+                path = Path(tempfile.mkdtemp()) / workflow.name
+
+            absolute_path = Path(get_absolute_path(path))
+            if absolute_path.exists():
+                communication.confirm(f"Output file '{path}' exists. Overwrite it?", abort=True, warning=True)
+
+            return absolute_path
+
+        output_path = get_output_path()
+
+        if isinstance(workflow, CompositePlan):
+            RenkuWorkflowFileExporter.convert_composite_plan(plan=workflow, path=output_path)
+        else:
+            RenkuWorkflowFileExporter.convert_plan(plan=workflow, path=output_path)
+
+        return output_path.read_text()
+
+    @staticmethod
+    def convert_composite_plan(plan: CompositePlan, path: Path):
+        """Converts a composite plan to a workflow file."""
+        data: Dict[str, Any] = {"name": plan.name}
+        if plan.description:
+            data["description"] = plan.description
+        if plan.keywords:
+            data["keywords"] = plan.keywords
+        if plan.plans:
+            data["steps"] = [{p.name: RenkuWorkflowFileExporter.convert_step(step=p)} for p in plan.plans]
+
+        write_yaml(path=path, data=data)
+
+    @staticmethod
+    def convert_plan(plan: Plan, path: Path):
+        """Converts a Plan to Renku workflow file format."""
+        # NOTE: There is no root plan and we need to create one; its name can't be the same as step's (i.e. plan's) name
+        data = {
+            "name": f"{plan.name}-workflow-file",
+            "steps": [{plan.name: RenkuWorkflowFileExporter.convert_step(step=plan)}],
+        }
+        write_yaml(path=path, data=data)
+
+    @staticmethod
+    def convert_step(step: Plan):
+        """Converts a Plan to workflow file dict."""
+
+        def convert_path(parameter):
+            data = {"path": parameter.actual_value}
+            if parameter.prefix:
+                data["prefix"] = parameter.prefix.rstrip(" =")
```

Comment:
We don't care if `=` is there or not when executing a workflow file, both forms are supported. However, I've added back `=` since its absence might cause confusion. Whitespaces are stripped though.